src/resources/resources.ts

```diff
@@ -0,0 +1,252 @@
+    }
+    if (Object.prototype.hasOwnProperty.call(apikeyObject, "authorizer")) {
+      const authorizer: Record<string, Array<unknown>> = {};
+      authorizer[
+        JSON.parse(
+          JSON.stringify(
+            (<Record<string, string>>apikeyObject.authorizer).authorizerName
+          )
+        )
+      ] = [];
+      security.push(authorizer);
+    }
+  }
+
+  const configObjects = <Array<IcurdObject>>config.objects;
+  configObjects.map((data: IcurdObject) => {
+    const pathName = data["path"];
+    swaggerPaths[pathName] = attachMethods(data["methods"], data, security);
+    return null;
+  });
+
+  swagger["paths"] = swaggerPaths;
+  const doc = new yaml.Document();
+  doc.contents = swagger;
+  utilities.writeFile(<string>config["filepath"], doc.toString());
+}
+
+const attachMethods = (
+  methodArray: Array<string>,
+  data: IcurdObject,
+  security: Array<Record<string, Array<unknown>>>
+) => {
+  const result: Record<string, object> = {};
+  if (methodArray.length) {
+    methodArray.map((item: string) => {
+      result[item] = JSON.parse(
+        JSON.stringify(configs.SwaggerPathSkeleton[item])
+      );
+      if (item !== "options") {
+        let uri =
+          (<Record<string, Record<string, Record<string, unknown>>>>(
+            result[item]
+          ))["x-amazon-apigateway-integration"]["uri"]["Fn::Sub"] +
+          configs.APIGatewayURI[data["resourcetype"]];
+
+        if (data["resourcetype"] == "lambda") {
+          uri = uri.replace("lambda.Arn", data["resource"] + ".Arn");
+        }
+
+        if (security !== undefined && Object.keys(security).length > 0) {
+          (<Record<string, unknown>>result[item])["security"] = JSON.parse(
+            JSON.stringify(security)
+          );
+        }
+        (<Record<string, Record<string, Record<string, unknown>>>>result[item])[
+          "x-amazon-apigateway-integration"
+        ]["uri"]["Fn::Sub"] = uri;
+      }
+      (<Record<string, Record<string, unknown>>>result[item])[
+        "x-amazon-apigateway-integration"
+      ]["credentials"] = {};
+      (<Record<string, Record<string, Record<string, unknown>>>>result[item])[
+        "x-amazon-apigateway-integration"
+      ]["credentials"]["Fn::Sub"] = "${" + data["role"] + ".Arn}";
+      return null;
+    });
+  }
+
+  return result;
+};
+
+export function apigatewaypath(template: ISAMTemplateResource, path: string) {
+  const definationbody = JSON.parse(JSON.stringify(configs.APIGatewaySkeleton));
+  definationbody["Fn::Transform"]["Parameters"]["Location"] = path;
+  template["Properties"]["DefinitionBody"] = definationbody;
+  return template;
+}
+export const resourceGeneration = function (
+  resource_name: string,
+  config: Record<string, unknown>
+) {
+  const resource_properties = JSON.parse(
+    JSON.stringify(configs.AWSResources[resource_name])
+  );
+  let template: ISAMTemplateResource = <ISAMTemplateResource>{};
+  for (const j in resource_properties.attributes) {
```

Comment:
this is a common change that would be good to have. It is better to use `of` instead of `in` for looping variables for an array of object as the loop variable will have the value of the object instead of the index of that object in the array.